(가윤) 1st week hw pr

farfromnow opened the pull request:
hw/week1 안에있음

#title에서 padding적용을 percentage로 했더니 background-image 사이즈가 변하는데 해결해줘...

categorize만 했고 font family만 정해둠
지금은 title부터 만지는 단계

---

기본정보
    이름
    사진
    학과
    연락처
        인스타그램
        이메일
        페이스북

관심분야
    ~~

hanyundo commented:
첫 숙제 PR 수고 하셨습니다! 😄 
몇가지 조언을 드릴게요.

1. txt/a.txt 는 현재 master 브랜치에서도 사라졌습니다. 해당 폴더(txt)를 삭제해주시구요

2. 일단 지금 background 사진이 화면에 베젤(?) 없이 딱 맞게 들어가게 하고 싶었을 것 같아요.
body 태그의 기본 속성이 margin 값이 있기 때문에 지금 딱 맞게 안 들어가고 있거든요
따라서, 이렇게 해주시면 될 것 같습니다.
```css
body {
  margin: 0;
}
```

3. padding 속성은 오브젝트의 길이를 늘어나게 합니다. 
그래서 ```#title```의 ```height: 100vh```였음에도 불구하고 조금 더 밑으로 내려오게 되었죠.
그에 따라 배경 사진도 더 커지게 되었지요.
padding과 margin은 처음에 가장 헷갈리는 개념이기도 합니다. 다음 수업 시간에도 자세히 알아볼 거구요.
일단 http://pungwun.blog.me/220239716987 이 페이지를 참고해서 수정해보도록 해요.
개인적인 생각으로는 ```#title``` 내부에 ```#brand```를 만들고 ```#brand```의 margin-top 값을 주는 게 의도하는 바와 가장 비슷하지 않았을까 싶습니다.

4. ```#infoList``` 가 두 개가 있죠 현재. id 속성은 딱 한 개의 태그에만 주길 바래요.
게다가 두 번째 ```#infoList```는 제대로 태그가 닫히지 않았어요. ```</div>```를 꼭 감싸주길 바랍니다.

5. CSS 파일의 ```#title```의 닫는 중괄호가 Indent에 맞지 않는 것 같아요. 

6. CSS 파일을 분리한 점, 웹 폰트를 사용한 점 등 배우지 않은 부분을 스스로 찾아서 해내서 정말 잘했어요 👍

farfromnow commented:
1. 정돈
```#title```에는 background 요소만, ```#brand```에는 font 요소만 들어가도록 분리
```#brand```를 ```#title```하위 id로 가게끔 하고 padding 피드백을 받아들여 수정
```#brand```에서 원하던 효과가 나지 않아서 vh에서 vm으로 수정
```#infoList``` 하위 class인 ```.introduce_title```과 ```.introduce_p``` 중복 속성을 빼서 ```#infoList```로 옮김
indent, </> 정돈

2. ```fixed``` 관련
https://brunch.co.kr/@careertrack/10 이와 비슷한 효과를 주기 위해 ```fixed```를 적용
background가 존재하는 ```#title```에는 ```background_attachment: fixed;```를 적용
그러나 ```#brand```에 ```position: fixed;```를 적용하자 ```#brand```는 살아남고 ```#title```은 사라지는 현상 발생 (흰 글씨라 잘 안 보이지만 살아있음)

hanyundo commented:
1) ```#brand```에게 ```position: fixed``` 속성을 주게 되면 width가 제멋대로 될 거에요.
```width: 100%```도 동시에 주시면 됩니다. 그러면 텍스트가 스크롤에 의해 움직이지 않게 될 거에요.

2) **z-index** 라는 개념을 알고 있어야 돼요. z-index는 정수 값을 줄 수 있는데, Photoshop의 Layer와 같은 개념이라고 생각하면 쉽습니다. z-index 값이 높을 수록 상단에 위치하게 됩니다. 
다만, 일반적인 div는 background가 비어있기 때문에 ```#infoList```가 스크롤에 의해 올라오게 되어도 텍스트를 덮지 않게 됩니다. (텍스트랑 배경 모두 흰색이라 잘 안 보일 수 있는데 색깔을 바꿔서 한번 테스트해보면 글자가 제대로 가려지는지 안 가려지는지 확인할 수 있을 겁니다.)
따라서, ```#infoList```에게 ```background-color: white```를 설정하도록 합시다. 그러면 스크롤이 내려가면서 글자를 ```#infoList```가 가리게 됩니다.
그런데, ```background-color```는 margin에 해당하는 부분에 적용이 되지 않습니다. 따라서 default margin 값을 가지고 있는 ```#infoList```의 아주 조그마한 부분은 텍스트가 또 안 가려질 거에요. 따라서 ```#infoList```의 속성을 ```margin-top: 0``` 이렇게 해줍시다.
**중요한 것은, z-index는 position 속성이 설정 되어 있는 태그들에게만 적용이 됩니다.** 따라서,
```#title```에게 ```position: relative``` 속성을 주고, ```#infoList```에게 ```position: relative``` 속성을 줍니다. ```#brand```는 ```position: fixed```가 이미 되어있겠죠?
아래 링크를 한번 읽어보세요.
https://developer.mozilla.org/ko/docs/Web/CSS/Understanding_z-index/Adding_z-index

farfromnow commented:
1. ```#title``` 관련 피드백 모두 반영
2. 이미지 추가

3. **이미지 크롭**
이미지 크롭을 다음과 같이 했는데요,
```css
body {
<img src="https://c1.staticflickr.com/8/7342/14055745934_d178026dce_b.jpg" alt="photography" title="photography" style="object-fit: cover; width: 70vh;">
}
```
무조건 이미지 원본 중앙 기준이라 ```left: 30%;```를 적용해 봤는데 결과가 똑같아 삭제했습니다.
방법이 있을까요? (물론 있겠지..)

4. 이미지들을 ```imgArr``` class를 통해 세로 높이를 제한하였는데,
사실 원하는 건 ```**같은 줄에 놓인 이미지들의 높이는 같되, 한 줄의 이미지 너비와 여백 합이 모두 같도록**``` 하는 것입니다.
css에서 미지수를 지정할 수 없는 것으로 아는데, JavaScript를 써야 한다든가... 하는 문제인가요?

hanyundo commented:
👍 수고하셨습니다.
제가 드리고 싶은 몇 가지 조언은 아래와 같습니다.

---

1) ```.introduction_title``` 태그 내에 ```.introduction_p```가 있는데 살짝 좋은 구조 같진 않습니다.
```.introduction``` 안에 children으로 ```.introduction_title```, ```.introduction_p```가 있는 게 좋을 것 같아요. 나중에 CSS 선택자나 jQuery 같은 걸 사용할 때 태그 사이의 계층 구조는 코드 직관성을 가지기 위해 매우 중요하게 작용하게 될 겁니다.

2) ```-```를 사용하는 것 보다 ```<hr>``` 같은 태그를 사용하는 것을 추천드립니다. 그렇게 중요하진 않습니다. HTML 내의 텍스트는 최대한 디자인적 요소를 줄이는 것이 좋다고 합니다.

3) CSS로 이미지를 Crop하는 것은 공식적으로는 없는 것 같습니다. 아마도 꼼수로서 해결하는 것 같은데 여기에 몇 가지 시도를 소개해놨으니 참고하길 바랍니다. 
https://themetry.com/css-image-cropping/
제 개인적인 생각으로는 ```<div>``` 안에 ```<img>```를 넣고 ```<div>```의 ```overflow: hidden```으로 두고 ```<img>```의 ```margin``` 값을 음수 등을 둬서 조절해서 표시하게 할 것 같아요. 될지 안 될지는 잘 모르겠습니다.
